### DISCUSSION TITLE
support for spring cli / initializr website or as dotnet new IKVM template

### DISCUSSION THREAD

**Discussion by jkone27** [ikvm-revived/ikvm-sdk-maven#56] [Ideas]
https://start.spring.io/

would be nice to have a dotnet tool like dotnet-ikvm-initializr that could init projects specifying maven dependencies from official or unofficial sources, and build the .NET solution

* https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-new

 * https://docs.spring.io/spring-boot/docs/current/reference/html/cli.html (official CLI) (a custom public server url can also be passed in)

**Comment by wasabii** [comment]
Neat idea. Probably would be separate from IKVM and IKVM maven though. A separate spring CLI tool. With a new template.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: jkone27 ***@***.***>
Sent: Monday, February 26, 2024 11:05:22 AM
To: ikvmnet/ikvm-maven ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ikvmnet/ikvm-maven] support for spring cli / initializr website or as dotnet new IKVM template (Issue #55)


https://start.spring.io/

would be nice to have a dotnet tool like dotnet-ikvm-initializr that could init projects specifying maven dependencies from official or unofficial sources, and build the .NET solution

  *   https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-new

  *   https://docs.spring.io/spring-boot/docs/current/reference/html/cli.html (official CLI) (a custom public server url can also be passed in)

**Comment by wasabii** [comment]
Going to move this to a discussion.

**Comment by wasabii** [comment]
So, the way I see it is there are a number of different possibilities to think through here.

First, IKVM provides a JDK image. So it can be a standin for a normal JDK.

So, running existing spring CLI tools on IKVM would be possible by just setting JAVA_HOME or whatever the existing spring CLI requires. There might be room for a dotnet global tool that just does that by default dotnet spring new Foo, etc. Existing spring tool, just running or compiled for IKVM.

Second, again, we're a JDK. Technically, gradle and Maven POM themselves could be run inside IKVM (if there aren't bugs). So, you could use the existing spring tools to generate an existing spring project, and just build/run it on IKVM.

Third, an entire Spring project template, targeting .NET, on MSBuild. This could probably be added to the existing spring CLI (looks like templates can be installed). But, it could also be added to Visual Studio and the .NET CLI templates. This could have both Java and C# versions. IKVM.NET.Sdk lets you build Java directly out of MSBuild.
